src/iio_wrapper.cpp

```diff
@@ -693,3 +735,162 @@ int32_t IIOWrapper::decimation_filter()
+    return ret;
+}
+
+int IIOWrapper::set_data_ready_polarity(int32_t val)
+{
+    long long valuel = val;
+    int ret = iio_device_debug_attr_write_longlong(m_dev,"data_ready_polarity", valuel);
+    return ret;
+}
+
+int IIOWrapper::set_sync_signal_scale(int32_t val)
+{
+    long long valuel = val;
+    int ret = iio_device_debug_attr_write_longlong(m_dev,"sync_signal_scale", valuel);
+    return ret;
+}
+
+int IIOWrapper::set_decimation_filter(int32_t val)
+{
+    long long valuel = val;
+    int ret = iio_device_debug_attr_write_longlong(m_dev,"decimation_filter", valuel);
+    return ret;
+}
+
+int IIOWrapper::set_accel_calibbias_x(int32_t val)
+{
+    long long valuel = val;
+    int ret = iio_channel_attr_write_longlong(m_channel_accel_x, "calibbias", valuel);
+    return ret;
+}
+
+int IIOWrapper::set_accel_calibbias_y(int32_t val)
+{
+    long long valuel = val;
+    int ret = iio_channel_attr_write_longlong(m_channel_accel_y, "calibbias", valuel);
+    return ret;
+}
+
+int IIOWrapper::set_accel_calibbias_z(int32_t val)
+{
+    long long valuel = val;
+    int ret = iio_channel_attr_write_longlong(m_channel_accel_z, "calibbias", valuel);
+    return ret;
+}
+
+int IIOWrapper::set_anglvel_calibbias_x(int32_t val)
+{
+    long long valuel = val;
+    int ret = iio_channel_attr_write_longlong(m_channel_anglvel_x, "calibbias", valuel);
+    return ret;
+}
+
+int IIOWrapper::set_anglvel_calibbias_y(int32_t val)
+{
+    long long valuel = val;
+    int ret = iio_channel_attr_write_longlong(m_channel_anglvel_y, "calibbias", valuel);
+    return ret;
+}
+
+int IIOWrapper::set_anglvel_calibbias_z(int32_t val)
+{
+    long long valuel = val;
+    int ret = iio_channel_attr_write_longlong(m_channel_anglvel_z, "calibbias", valuel);
+    return ret;
+}
+
+int IIOWrapper::set_sampling_frequency(int32_t val)
+{
+    long long valuel = val;
+    int ret = iio_device_attr_write_longlong(m_dev, "sampling_frequency", valuel);
+    return ret;
+}
+
+int32_t IIOWrapper::sampling_frequency()
+{
+    long long valuel;
+    iio_device_attr_read_longlong(m_dev,"sampling_frequency", &valuel);
+
+    return valuel;
+}
+
+int IIOWrapper::set_software_reset(int32_t val)
+{
+    int ret = iio_device_debug_attr_write_longlong(m_dev,"software_reset", 1);
+    return ret;
+}
+
+int IIOWrapper::set_flash_memory_test(int32_t val)
+{
+    int ret = iio_device_debug_attr_write_longlong(m_dev,"flash_memory_test", 1);
+    return ret;
+}
+
+int IIOWrapper::set_flash_memory_update(int32_t val)
+{
+    int ret = iio_device_debug_attr_write_longlong(m_dev,"flash_memory_update", 1);
+    return ret;
+}
+
+int IIOWrapper::set_sensor_self_test(int32_t val)
+{
+    int ret = iio_device_debug_attr_write_longlong(m_dev,"sensor_self_test", 1);
+    return ret;
+}
+
+int IIOWrapper::set_factory_calibration_restore(int32_t val)
+{
+    int ret = iio_device_debug_attr_write_longlong(m_dev,"factory_calibration_restore", 1);
+    return ret;
+}
```

Comment:
the APIs should not receive a pramater from the user, the APIs should look something like thos:
```
int IIOWrapper::perform_factory_calibration_restore_command()
{
    int ret = iio_device_debug_attr_write_longlong(m_dev,"factory_calibration_restore", 1);
    return ret;
}
```